### PR TITLE
upgrade runtime to v2 api and cleanup some types

### PR DESCRIPTION
**Pull request by bschwinn**

### CODE REVIEW

File: src/app.template.json [hunk 1]
```diff
@@ -16,6 +16,6 @@
     },
     "runtime": {
         "arguments": "--enable-aggressive-domstorage-flushing",
-        "version": "9.61.33.32"
+        "version": "beta"
```
Comment (MichaelMCoates) [inline src/app.template.json]: Do we want to tie this to beta or a specific version? What is this used for again?

File: src/demo/app-launcher.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     ],
     "runtime": {
         "arguments": "--v=1 --enable-crash-reporting",
-        "version": "9.61.33.32"
+        "version": "beta"
```
Comment (MichaelMCoates) [inline src/demo/app-launcher.json]: Do we want to tie this to beta or a specific version?

File: src/ui/components/NotificationGroup.tsx [hunk 3]
```diff
@@ -2,11 +2,10 @@ import * as React from 'react';
 import { NotificationType } from './NotificationType';
 import { INotification } from '../models/INotification';
 import { INotificationGroupProps } from '../models/INotificationGroupProps';
-import { ISenderInfo } from '../../provider/Models/ISenderInfo';
-import { Fin } from '../../fin';
 import { eGroupMethod } from './App';
 
-declare var fin: Fin;
+import {NotificationCenterAPI} from '../NotificationCenterAPI';
+declare var window: Window&{notifications: NotificationCenterAPI};
```
Comment (MichaelMCoates) [inline src/ui/components/NotificationGroup.tsx]: Do we want the API available on the window?

File: src/ui/index.ts [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
-import {Fin} from '../fin';
 import {ISenderInfo} from '../provider/Models/ISenderInfo';
 import {Notification} from '../Shared/Models/Notification';
-
-declare var fin: Fin;
+import {INotification} from './models/INotification';
+import {NotificationCenterAPI} from './NotificationCenterAPI';
+declare var window: Window&{notifications: NotificationCenterAPI};
 
 const IDENTITY = {
```
Comment (MichaelMCoates) [inline src/ui/index.ts]: Looks like we're using this in two places. Do we want a separate location for it so we don't have to worry about changing it in both spots?

File: src/ui/index.ts [hunk 5]
```diff
@@ -66,12 +66,18 @@ fin.desktop.main(async () => {
             const plugin = await pluginP;
             const payload = {uuid};
             const appNotifications = await plugin.dispatch('fetch-app-notifications', payload);
+            appNotifications.forEach(n => {
+                n.date = new Date(n.date);
+            });
             console.log('appNotifications', appNotifications);
         },
-        fetchAllNotifications: async () => {
+        fetchAllNotifications: async() => {
```
Comment (MichaelMCoates) [inline src/ui/index.ts]: Lint?

File: src/ui/index.ts [hunk 6]
```diff
@@ -87,7 +93,7 @@ fin.desktop.main(async () => {
             const success = await plugin.dispatch('clear-app-notifications', payload);
             console.log('success', success);
         },
-        addEventListener: async (event: string, cb: (payload: Notification|ISenderInfo, sender: ISenderInfo) => string) => {
+        addEventListener: async (event: string, cb: (payload: any, sender: ISenderInfo) => string) => {
```
Comment (MichaelMCoates) [inline src/ui/index.ts]: Any way we can type this?

File: src/ui/components/NotificationGroup.tsx (hunk 3 above)
Comment (bschwinn) [reply src/ui/components/NotificationGroup.tsx]: we switched to using real types so can't really "extend" fin anymore.

Comment (bschwinn) [reply src/ui/components/NotificationGroup.tsx]: I will add an "openfin" in there so it's `window.openfin.notifications`

File: src/ui/index.ts (hunk 6 above)
Comment (bschwinn) [reply src/ui/index.ts]: we may be able to reference a "shared" javascript file in both the UI and client, probably best for another smaller PR.

File: src/ui/index.ts (hunk 4 above)
Comment (bschwinn) [reply src/ui/index.ts]: we may be able to reference a "shared" javascript file in both the UI and client, probably best for another smaller PR.

File: src/ui/index.ts (hunk 5 above)
Comment (bschwinn) [reply src/ui/index.ts]: looks like UI isn't being linted, making a note, will do in another PR.

File: src/ui/index.ts (hunk 6 above)
Comment (bschwinn) [reply src/ui/index.ts]: yeah, lemme see, i think i may have been getting lazy.

File: src/ui/index.ts (hunk 5 above)
Comment (bschwinn) [reply src/ui/index.ts]: added src/ui/index.ts to tsconfig.json

File: src/ui/index.ts (hunk 6 above)
Comment (bschwinn) [reply src/ui/index.ts]: done and I think the types are better resolved now because of this.

**Pull request merged**